Build broken with GCC 9

GCC recently added a new `-Wredundant-move` warning enabled with `-Wall`, and since the build specifies `-Werror` it breaks unless you remove it:

```
[292/379] Building CXX object src/CMakeFiles/amber_unittests.dir/executor_test.cc.o
FAILED: src/CMakeFiles/amber_unittests.dir/executor_test.cc.o 
/usr/lib64/ccache/c++  -DAMBER_CTS_VULKAN_HEADER=0 -DAMBER_ENABLE_CLSPV=0 -DAMBER_ENABLE_DXC=0 -DAMBER_ENABLE_LODEPNG=1 -DAMBER_ENABLE_SHADERC=1 -DAMBER_ENABLE_SPIRV_TOOLS=1 -DAMBER_ENGINE_DAWN=0 -DAMBER_ENGINE_VULKAN=1 -I../include -I../ -I../third_party/spirv-tools/include -I../third_party/shaderc/libshaderc/include -I../third_party/glslang/SPIRV/.. -I../third_party/glslang/SPIRV/../External -I../third_party/glslang/glslang/.. -I../third_party/spirv-headers/include -isystem ../third_party/googletest/googlemock/include -isystem ../third_party/googletest/googlemock -isystem ../third_party/googletest/googletest/include -isystem ../third_party/googletest/googletest -g -fPIE   -Wno-global-constructors -Wno-weak-vtables -std=c++11 -fno-exceptions -fno-rtti -fvisibility=hidden -Wall -Werror -Wextra -Wno-padded -Wno-switch-enum -Wno-unknown-pragmas -pedantic-errors -pthread -std=gnu++11 -MD -MT src/CMakeFiles/amber_unittests.dir/executor_test.cc.o -MF src/CMakeFiles/amber_unittests.dir/executor_test.cc.o.d -o src/CMakeFiles/amber_unittests.dir/executor_test.cc.o -c ../src/executor_test.cc
../src/executor_test.cc: In member function ‘std::unique_ptr<amber::Engine> amber::vkscript::{anonymous}::VkScriptExecutorTest::MakeAndInitializeEngine(const std::vector<std::__cxx11::basic_string<char> >&, const std::vector<std::__cxx11::basic_string<char> >&, const std::vector<std::__cxx11::basic_string<char> >&)’:
../src/executor_test.cc:205:21: error: redundant move in return statement [-Werror=redundant-move]
  205 |     return std::move(engine);
      |            ~~~~~~~~~^~~~~~~~
../src/executor_test.cc:205:21: note: remove ‘std::move’ call
At global scope:
cc1plus: error: unrecognized command line option ‘-Wno-weak-vtables’ [-Werror]
cc1plus: error: unrecognized command line option ‘-Wno-global-constructors’ [-Werror]
cc1plus: all warnings being treated as errors
```
While something like `-Werror` is fine to have in CI, enabling it by default is a well-known antipattern in open-source software projects. Users and developers with a newer compiler than what is tested in CI will likely have the build fail due to new warnings when just following the instructions in the README, which makes it user-hostile.

--- a/src/executor_test.cc
+++ b/src/executor_test.cc
@@ -202,7 +202,7 @@ class VkScriptExecutorTest : public testing::Test {
     auto engine = MakeUnique<EngineStub>();
     engine->Initialize(nullptr, nullptr, features, instance_extensions,
                        device_extensions);
-    return std::move(engine);
+    return engine;
   }
   EngineStub* ToStub(Engine* engine) {
     return static_cast<EngineStub*>(engine);

--- a/src/executor_test.cc
+++ b/src/executor_test.cc
@@ -199,7 +199,7 @@ class VkScriptExecutorTest : public testing::Test {
       const std::vector<std::string>& features,
       const std::vector<std::string>& instance_extensions,
       const std::vector<std::string>& device_extensions) {
-    auto engine = MakeUnique<EngineStub>();
+    std::unique_ptr<Engine> engine = MakeUnique<EngineStub>();
     engine->Initialize(nullptr, nullptr, features, instance_extensions,
                        device_extensions);
     return engine;

